Promote LCW Beta to 1.0 Documentation

OWinfreyATL opened the pull request.

OWinfreyATL commented:
@FaithOmbongi 

Hey Faith. This is promoting LCW from BETA to v1.0. Vast majority of these articles are just an update move from beta to v1.0 with a few changes:

1. emailsettings.md added
2.customExtensionAuthenticationConfiguration.md is inherited by [azureAdPopTokenAuthentication] with v1.0 but still azureADTokenAuthentication with BETA
3.New custom example being added with  create new version for customized email in identitygovernance-workflow-createnewversion.md

Still a few weeks off, but moving to review now as API schema passed review.

FaithOmbongi commented:
> @FaithOmbongi
> 
> Hey Faith. This is promoting LCW from BETA to v1.0. Vast majority of these articles are just an update move from beta to v1.0 with a few changes:
> 
> 1. emailsettings.md added
>    2.customExtensionAuthenticationConfiguration.md is inherited by [azureAdPopTokenAuthentication] with v1.0 but still azureADTokenAuthentication with BETA
>    3.New custom example being added with  create new version for customized email in identitygovernance-workflow-createnewversion.md
> 
> Still a few weeks off, but moving to review now as API schema passed review.

Thanks for the headsup @OWinfreyATL. Can you create a review work item in my ADO? https://aka.ms/msgraph-identity-workitem

Thanks!

OWinfreyATL commented:
> > @FaithOmbongi
> > Hey Faith. This is promoting LCW from BETA to v1.0. Vast majority of these articles are just an update move from beta to v1.0 with a few changes:
> > 
> > 1. emailsettings.md added
> >    2.customExtensionAuthenticationConfiguration.md is inherited by [azureAdPopTokenAuthentication] with v1.0 but still azureADTokenAuthentication with BETA
> >    3.New custom example being added with  create new version for customized email in identitygovernance-workflow-createnewversion.md
> > 
> > Still a few weeks off, but moving to review now as API schema passed review.
> 
> Thanks for the headsup @OWinfreyATL. Can you create a review work item in my ADO? https://aka.ms/msgraph-identity-workitem
> 
> Thanks!

Done Here: https://dev.azure.com/msft-skilling/Content/_workitems/edit/63881 

thanks! :)

msewaweru reviewed (changes requested):
Hi @OWinfreyATL - This is batch 1 of the review. Please see main the highlights captured: 
1. GLOBAL: LifecycleWorkflowsContainer no longer inherits from entity v1.0. Remove all reference to this inheritance.
2. Remove tabs on all the HTTP examples.
3. authorizedApps property has been used extensively and is not in the schema. Please confirm its validity.
4. GLOBAL: Consider removing the Global Admin role from the LCW docs. There is an team initiative to stop document using it if there are lesser privileged roles.
5. GLOBAL: Casing for the role names should be capitalized for each word.
6. GLOBAL: Remove the tabs in all examples. These are inserted by the snippets generator.
7. GLOBAL: Seems inheritance from entity was removed in v1.0. Please remove these in 3 places the resource and the Id property descriptions and the JSON baseType representation.
8. Where I have highlighted missing properties, check beta side to update as well.

msewaweru reviewed (changes requested):
@OWinfreyATL, this is batch 2 and final part of cycle 1 of the review. Main highlights as below. Thanks.
1. Remove the preview tag from the beta TOC.
2. Remove the differentiation in the page titles as done for the beta docs.

OWinfreyATL commented on api-reference/v1.0/resources/identitygovernance-userprocessingresult.md:
```diff
@@ -0,0 +1,73 @@
+|:---|:---|:---|
+|[List userProcessingResults](../api/identitygovernance-workflow-list-userprocessingresults.md)|[microsoft.graph.identityGovernance.userProcessingResult](../resources/identitygovernance-userprocessingresult.md) collection|Get a list of the [userProcessingResult](../resources/identitygovernance-userprocessingresult.md) objects and their properties.|
+|[summary](../api/identitygovernance-userprocessingresult-summary.md)|[microsoft.graph.identityGovernance.userSummary](../resources/identitygovernance-usersummary.md)|Provides a summary of user processing results for a specified time period.|
+|[List taskProcessingResults](../api/identitygovernance-userprocessingresult-list-taskprocessingresults.md)|[microsoft.graph.identityGovernance.taskReport](../resources/identitygovernance-taskprocessingresult.md) collection|Get a list of the [taskProcessingResult](../resources/identitygovernance-taskprocessingresult.md) objects and their properties.|
```
We had a discussion around this and we decided it would remain under the userprocessingresults section as the call goes through user processing(the call checks for the task processing results of a user processing result).

OWinfreyATL commented on api-reference/v1.0/resources/identitygovernance-run.md:
```diff
@@ -0,0 +1,80 @@
+|[List runs](../api/identitygovernance-workflow-list-runs.md)|[microsoft.graph.identityGovernance.run](../resources/identitygovernance-run.md) collection|Get a list of the [run](../resources/identitygovernance-run.md) objects and their properties.|
+|[Get run](../api/identitygovernance-run-get.md)|[microsoft.graph.identityGovernance.run](../resources/identitygovernance-run.md)|Read the properties and relationships of a [run](../resources/identitygovernance-run.md) object.|
+|[summary](../api/identitygovernance-run-summary.md)|[microsoft.graph.identityGovernance.runSummary](../resources/identitygovernance-runsummary.md)|Get a summary of workflows runs.|
+|[List user processing results](../api/identitygovernance-run-summary.md)|[microsoft.graph.identityGovernance.userProcessingResult](../resources/identitygovernance-userprocessingresult.md) collection|Get a list of user processing results of a run.|
+|[Get user processing result](../api/identitygovernance-run-summary.md)|[microsoft.graph.identityGovernance.userProcessingResult](../resources/identitygovernance-userprocessingresult.md)|Get a user processing result of a run|
+|[List task processing results of a run](../api/identitygovernance-run-list-taskprocessingresults.md)|[microsoft.graph.identityGovernance.taskReportSummary](../resources/identitygovernance-taskprocessingresult.md)|List task processing results from a run.|
```
As noted above this was decided to be listed under run as the call lists the task processing results of a workflow run.

OWinfreyATL commented on api-reference/v1.0/resources/identitygovernance-run.md:
```diff
@@ -0,0 +1,80 @@
+|[Get run](../api/identitygovernance-run-get.md)|[microsoft.graph.identityGovernance.run](../resources/identitygovernance-run.md)|Read the properties and relationships of a [run](../resources/identitygovernance-run.md) object.|
+|[summary](../api/identitygovernance-run-summary.md)|[microsoft.graph.identityGovernance.runSummary](../resources/identitygovernance-runsummary.md)|Get a summary of workflows runs.|
+|[List user processing results](../api/identitygovernance-run-summary.md)|[microsoft.graph.identityGovernance.userProcessingResult](../resources/identitygovernance-userprocessingresult.md) collection|Get a list of user processing results of a run.|
+|[Get user processing result](../api/identitygovernance-run-summary.md)|[microsoft.graph.identityGovernance.userProcessingResult](../resources/identitygovernance-userprocessingresult.md)|Get a user processing result of a run|
```
Updated.

OWinfreyATL commented on api-reference/v1.0/resources/identitygovernance-workflow.md:
```diff
@@ -0,0 +1,99 @@
+|[activate](../api/identitygovernance-workflow-activate.md)|None|Run a workflow on-demand.|
+|[Create workflowVersion](../api/identitygovernance-workflow-createnewversion.md)|[microsoft.graph.identityGovernance.workflowVersion](../resources/identitygovernance-workflowversion.md)|Create a new workflowVersion object.|
+|[restore](../api/identitygovernance-workflow-restore.md)|[microsoft.graph.identityGovernance.workflow](../resources/identitygovernance-workflow.md)|Restore a deleted workflow.|
+|[List runs](../api/identitygovernance-workflow-list-runs.md)|[microsoft.graph.identityGovernance.run](../resources/identitygovernance-run.md) collection|Get a list of the [runs](../resources/identitygovernance-run.md) for the workflow.|
+|[Get task](../api/identitygovernance-task-get.md)|[microsoft.graph.identityGovernance.task](../resources/identitygovernance-task.md) collection|Get the properties of a [task](../resources/identitygovernance-task.md) in the workflow.|
+|[List tasks](../api/identitygovernance-workflow-list-task.md)|[microsoft.graph.identityGovernance.task](../resources/identitygovernance-task.md) collection|Get a list of [tasks](../resources/identitygovernance-task.md) objects in the workflow.|
+|[List versions](../api/identitygovernance-workflow-list-versions.md)|[microsoft.graph.identityGovernance.workflowVersion](../resources/identitygovernance-workflowversion.md) collection|Get a list of [workflow versions](../resources/identitygovernance-workflowversion.md) of the workflow.|
```
These are present in the subset resource pages and nodes, but as these nodes are a subset of the workflow we had them listed here.

OWinfreyATL commented on api-reference/v1.0/resources/identitygovernance-customtaskextensioncallbackconfiguration.md:
```diff
@@ -0,0 +1,48 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|timeoutDuration|Duration| Callback time out in ISO 8601 time duration. Accepted time durations are between five minutes to three hours. For example, PT5M for five minutes and PT3H for three hours. Inherited from [customExtensionCallbackConfiguration](../resources/customextensioncallbackconfiguration.md).|
+|authorizedApps|microsoft.graph.application collection| Unique identifier for the application object. Inherited from [customExtensionCallbackConfiguration](../resources/customextensioncallbackconfiguration.md). |
```
I believe this is being added with another schema. Keeping it here for now, but will remove from the public preview refresh PR that will incorporate feedback from here.

OWinfreyATL commented:
@msewaweru Hey Eunice. Good feedback. I have addressed all of them(pending authorizedAPPs here).

This feature will have another round of public preview with these changes(BETA only) which I have incorporated into a PR here: https://github.com/microsoftgraph/microsoft-graph-docs/pull/20510

As that one will be the active one that will be with changes deploying this week I am not sure if I should request a look at the changes here or there :)

thanks again!

OWinfreyATL commented on api-reference/v1.0/resources/customextensioncallbackconfiguration.md:
```diff
@@ -0,0 +1,47 @@
+|Property|Type|Description|
+|:---|:---|:---|
+|timeoutDuration|Duration|Callback time out in ISO 8601 time duration. Accepted time durations are between five minutes to three hours. For example, PT5M for five minutes and PT3H for three hours.|
+|authorizedApps|microsoft.graph.application collection| Unique identifier for the application object.|
```
I have confirmed it was added in this schema change: https://microsoftgraph.visualstudio.com/onboarding/_git/onboarding/pullrequest/12515?_a=files

msewaweru commented:
> @msewaweru Hey Eunice. Good feedback. I have addressed all of them(pending authorizedAPPs here).
> 
> This feature will have another round of public preview with these changes(BETA only) which I have incorporated into a PR here: #20510
> 
> As that one will be the active one that will be with changes deploying this week I am not sure if I should request a look at the changes here or there :)
> 
> thanks again!

@OWinfreyATL - thanks for the good job on the updates. I'm still learning the process here and would like to hear what @FaithOmbongi thinks on the second PR. She will be back Monday next week and we can agree on a way forward then. Thanks.
